tests/tensor_search/test_bulk_search.py

```diff
@@ -0,0 +1,1685 @@
+import copy
```

Comment:
feedback: more tests to ensure correctness for more than 1 query. There are cases during bulk search processing where queries components are combined and then distributed out. We need to cover and test these cases

Reply:
I'm going to convert each of these single query tests, to a single bulk_search call with multiple queries.